plugins/volume/volumeset.go

```diff
@@ -38,28 +51,24 @@ type volumeSet struct {
 func (vs *volumeSet) Prepare(context *runtime.TaskContext) error {
 	vs.context = context
 
-	for volumeType, plugin := range vs.vm.plugins {
-		switch volumeType {
-		case "persistent":
-			for _, vol := range vs.payload.Persistent {
-				opts := &volumeOptions{
-					taskID:     context.TaskInfo.TaskID,
-					runID:      context.TaskInfo.RunID,
-					name:       vol.Name,
-					mountPoint: vol.MountPoint,
-				}
-				v, err := plugin.newVolume(opts)
-				if err != nil {
-					return err
-				}
-				vs.volumes = append(vs.volumes, v)
-			}
-		default:
+	for _, volume := range vs.payload {
+		plugin, exists := vs.vm.plugins[volume.Type]
+		if !exists {
 			panic(
 				"Unrecognized volume type. This should never happen and " +
 					"is considered a fatal error.",
 			)
 		}
+		opts := &volumeOptions{
+			taskID: context.TaskInfo.TaskID,
+			runID:  context.TaskInfo.RunID,
+			spec:   spec(volume),
+		}
+		v, err := plugin.newVolume(opts)
```

Comment:
maybe implement opts as a "map[string]interface{}" and then the plugin will do what it needs to with the options.